index.js

```diff
@@ -33,15 +33,16 @@ hexo.config.index_generator = Object.assign({
 hexo.extend.generator.register('index-i18n', function indexI18nGenerator(locals) {
   const config = this.config;
   const posts = locals.posts.sort(config.index_generator.order_by);
+  const path = config.index_generator.path || '';
   const languages = [].concat(config.language || [])
     .filter(lang => lang !== 'default');
   const defaultLanguage = languages[0];
   let indexPages = [].concat.apply([],
-    languages.map(lang => getIndexPages(lang, lang, posts, config))
+    languages.map(lang => getIndexPages(lang + '/' + path, lang, posts, config))
```

Comment:
Perhaps, you do need [path.join](https://nodejs.org/api/path.html#path_path_join_paths)?

```js
const path = require('path'); // At the top of the module.

/* In the function. */
const indexPath = _.get(config, 'index_generator.path', '');
getIndexPages(path.join(lang, indexPath), lang, posts, config); // Inside the `map`.
```